libs/coin-modules/coin-evm/src/staking/fetchers.ts

```diff
@@ -0,0 +1,153 @@
+import { log } from "@ledgerhq/logs";
+import { CryptoCurrency } from "@ledgerhq/types-cryptoassets";
+import { Stake } from "@ledgerhq/coin-framework/api/types";
+import { withApi } from "../network/node/rpc.common";
+import type { StakeCreate, StakingContractConfig, StakingStrategy } from "../types/staking";
+import { extractSeiDelegation, getSeiDelegationAmount, getCeloAmount } from "../utils";
+import { encodeStakingData, decodeStakingResult } from "./encoder";
+import { buildTransactionParams } from "./transactionData";
+import { getValidators } from "./validators";
+
+/**
+ * Generic staking fetcher that adapts to different blockchain requirements
+ */
+const createStakingFetcher = (
+  getValidatorsFn: (config: StakingContractConfig, currency: CryptoCurrency) => Promise<string[]>,
+) => {
+  return async (
+    address: string,
+    config: StakingContractConfig,
+    currency: CryptoCurrency,
+  ): Promise<Stake[]> => {
+    const validators = await getValidatorsFn(config, currency);
+    const logPrefix = currency.id === "sei_network_evm" ? "SEI" : "CELO";
+    return getStakesForValidators(address, config, currency, validators, logPrefix);
+  };
+};
+
+export const STAKING_CONFIG: Record<string, StakingStrategy> = {
+  sei_network_evm: {
+    fetcher: createStakingFetcher(async (_config, currency) => await getValidators(currency.id)),
+  },
+  celo: {
+    fetcher: createStakingFetcher(async (config, _currency) => [config.contractAddress]),
+  },
+};
+
+const AMOUNT_EXTRACTORS = {
+  sei_network_evm: (decoded: unknown): bigint => {
+    const delegation = extractSeiDelegation(decoded);
+    return getSeiDelegationAmount(delegation);
+  },
+  celo: getCeloAmount,
+};
+
+const getAmountFromDecoded = (currencyId: string, decoded: unknown): bigint => {
+  const extractor = AMOUNT_EXTRACTORS[currencyId as keyof typeof AMOUNT_EXTRACTORS];
+  return extractor ? extractor(decoded) : 0n;
+};
```

Comment:
Why is `STAKING_CONFIG` typed, and not `AMOUNT_EXTRACTORS ?
```suggestion
export const STAKING_CONFIG: Record<string, StakingStrategy> = {
  sei_network_evm: {
    fetcher: createStakingFetcher(async (_config, currency) => await getValidators(currency.id)),
  },
  celo: {
    fetcher: createStakingFetcher(async (config, _currency) => [config.contractAddress]),
  },
};

const AMOUNT_EXTRACTORS: Record<string, StakingExtractor> = { // StakingExtractor to create
  sei_network_evm: (decoded: unknown): bigint => {
    const delegation = extractSeiDelegation(decoded);
    return getSeiDelegationAmount(delegation);
  },
  celo: getCeloAmount,
};

const getAmountFromDecoded = (currencyId: string, decoded: unknown): bigint => {
  const extractor = AMOUNT_EXTRACTORS[currencyId];
  return extractor ? extractor(decoded) : 0n;
};
```